extensions/wikia/Email/Controller/WallMessageController.class.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\Check;
+use Email\EmailController;
+
+class WallMessageController extends EmailController {
+
+	private $messageUrl;
+	private $messageTitle;
+	private $messageBody;
+
+	private $wallUserName;
+	private $senderUserName;
+	private $receiverUserName;
+
+	public function initEmail() {
+		$this->wallUserName = $this->request->getVal( 'wallUserName' );
+		$threadUserName = $this->request->getVal( 'threadUserName' );
+		$replyUserName = $this->request->getVal( 'replyUserName' );
+		$isThread = $this->request->getVal( 'isThread' );
+
+		if ( $isThread ) {
+			$this->senderUserName = $threadUserName;
+			$this->receiverUserName = $this->wallUserName;
+		} else {
+			$this->senderUserName = $replyUserName;
+			$this->receiverUserName = $threadUserName;
+		}
+
+		$this->assertUserRolesSet();
+
+		$this->messageUrl = $this->request->getVal( 'messageUrl' );
+		$this->messageTitle = $this->request->getVal( 'messageTitle' );
+		$this->messageBody = $this->request->getVal( 'messageBody' );
+
+		$this->assertMessageDetails();
+	}
+
+	private function assertUserRolesSet() {
+		if ( empty( $this->senderUserName ) ) {
+			throw new Check( "Could not determine message sender" );
+		}
+
+		if ( empty( $this->receiverUserName ) ) {
+			throw new Check( "Could not determine message recipient" );
+		}
+	}
+
+	private function assertMessageDetails() {
+		if ( empty( $this->messageUrl ) ) {
+			throw new Check( "Message URL parameter (messageUrl) is required" );
+		}
+
+		if ( empty( $this->messageTitle ) ) {
+			throw new Check( "Message title parameter (messageTitle) is required" );
+		}
+
+		if ( empty( $this->messageBody ) ) {
+			throw new Check( "Message body parameter (messageBody) is required" );
+		}
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->senderUserName,
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'detailsSubject' => $this->messageTitle,
+			'details' => $this->messageBody,
+			'buttonText' => $this->getButtonText(),
+			'buttonLink' => $this->messageUrl,
+			'contentFooterMessages' => [
+				$this->getRecentMessagesText()
+			],
+		] );
+	}
+
+	/**
+	 * Whether this thread topic or thread reply is meant for the user getting this email
+	 *
+	 * @return boolean
+	 */
+	protected function isForTargetUser() {
+		$targetName = $this->targetUser->getName();
+		return $targetName == $this->receiverUserName;
+	}
+
+	/**
+	 * Get the email subject line
+	 *
+	 * @return string
+	 */
+	public function getSubject() {
+		if ( $this->isForTargetUser() ) {
```

Comment:
I see couple conditions of this type in this class. What do you think about creating separate classes to handle those cases?